### PR TITLE
fix for tags update and delete op

### PR DESCRIPTION
**Pull request by LochanRn**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Allows aks tags to be updated and deleted 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
allows aks tags to be updated and deleted.
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
@LochanRn thanks for looking into this! You'll have to update test cases here to deal w/ new `Tags` property: `azure/services/managedclusters/managedclusters_test.go`

**Comment by LochanRn** [comment]
hey @jackfrancis thanks, will update the PR accordingly :)

**Comment by LochanRn** [comment]
@jackfrancis can you please review this PR again, sorry for the delay.

File: azure/services/managedclusters/managedclusters.go [hunk 1]
```diff
@@ -240,6 +241,10 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		},
 	}
 
+	if tags := *to.StringMapPtr(managedClusterSpec.Tags); len(tags) != 0 {
```
Comment (jackfrancis) [inline azure/services/managedclusters/managedclusters.go:244]: Can you explain why this change (only add tags property value if it is present in ManagedCluster spec) allows tags to be updated?

Comment (LochanRn) [reply azure/services/managedclusters/managedclusters.go:244]: No it's because when tags are empty the existing object returned from the azure api has a nil value for tags but the current object has map[string]string with length 0, the diff function we use is returning true for this case, which triggers unnecessary updates.

**Review by jackfrancis — APPROVED** [review]
/lgtm
